src/xref2/env.ml

```diff
@@ -356,6 +357,12 @@ let add_extension_constructor identifier
   add_to_elts Kind_Extension identifier (`Extension (identifier, ec)) env
   |> add_cdocs identifier ec.doc
 
+let add_extension_decl identifier (ec : Component.Extension.Constructor.t) env =
+  add_to_elts Kind_ExtensionDecl identifier
+    (`ExtensionDecl (identifier, ec))
+    env
+  |> add_cdocs identifier ec.doc
+
```

Comment:
I don't think adding `extension_decl` to the environment is the way to go: we already have the information from the `extension`s added to the environment!

Reply:
done